public/app/plugins/datasource/loki/LanguageProvider.test.ts

```diff
@@ -57,11 +57,14 @@ describe('Language completion provider', () => {
 
     it('should again fetch labels on second start with different timerange', async () => {
       const languageProvider = new LanguageProvider(datasource);
+      jest.mocked(datasource.getTimeRangeParams).mockRestore();
       const fetchSpy = jest.spyOn(languageProvider, 'fetchLabels').mockResolvedValue([]);
       await languageProvider.start();
       expect(fetchSpy).toHaveBeenCalledTimes(1);
       await languageProvider.start(mockTimeRange);
       expect(fetchSpy).toHaveBeenCalledTimes(2);
+      await languageProvider.start(mockTimeRange);
+      expect(fetchSpy).toHaveBeenCalledTimes(2);
```

Comment:
I'm a bit confused as to what this is accomplishing, won't this test pass without any of the implementation in this PR?

Reply:
Good question. I pass the same time range again, and the expectation is that it sees that it was already started at that time and it doesn't get called 3 times.

Reply:
gotcha, so it's good to keep this in here, but we should probably add a unit test that shows that changing the time range and triggering another query will update the time range sent?

Reply:
It's not a bad idea, but it actually uses the time range that you pass as an argument, that you can find in all the functions: https://github.com/grafana/grafana/blob/main/public/app/plugins/datasource/loki/LanguageProvider.ts#L253 , or the default time range, which even the `start()` method uses.

This whole `start()` thing is just to keep track of "when" it was started so it doesn't "restart".

The actual regression test would be to change the time range in `MonacoQueryField` and assert that `LanguageProvider` was updated, but we don't unit test the editor.